packages/api/src/folder/service.ts

```diff
@@ -0,0 +1,69 @@
+import createError from "http-errors";
+import { logger } from "@stela/logger";
+import { db } from "../database";
+import type {
+  FolderColumnsForUpdate,
+  FolderRow,
+  PatchFolderRequest,
+} from "./models";
+import { requestFieldsToDatabaseFields } from "./helper";
+
+export const getFolderById = async (requestQuery: {
+  folderId: string;
+}): Promise<FolderRow> => {
+  const folder = await db
+    .sql<FolderRow>("folder.queries.get_folder_by_id", {
+      folderId: requestQuery.folderId,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError(
+        "failed to retrieve featured archives"
+      );
+    });
+
+  if (folder.rows[0] === undefined) {
+    throw new createError.NotFound("Folder not found.");
+  }
+
+  return folder.rows[0];
+};
+
+function buildPatchQuery(patchFolderRequest: FolderColumnsForUpdate): string {
+  const updates = Object.entries(patchFolderRequest)
+    .filter(([key, value]) => value !== undefined && key !== "folderId")
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE folder
+    SET ${updates.join(", ")}
+    WHERE folderid = :folderId
+    RETURNING folder.folderid AS "folderId"
+  `;
+
+  return query.trim();
+}
+
+export const patchFolder = async (
```

Comment:
This function needs a mechanism to ensure the authenticated user is updating a folder they have access to

Reply:
And we missed this on PATCH /record, so here's a ticket to fix that https://permanent.atlassian.net/browse/PER-9942

Reply:
After we validate the access check for folders, I'll pick up this one